server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -977,6 +1045,175 @@ private void insertPendingSegmentIntoMetastore(
           .execute();
   }
 
+  /**
+   * Allocates and returns any extra versions that need to be committed for the
+   * given append segments.
+   * <p>
+   * This is typically needed when a REPLACE task started and finished after
+   * these append segments had already been allocated. As such,
+   * there would be some used segments in the DB with versions higher than these
+   * append segments.
+   */
+  private Set<DataSegment> getSegmentsToUpgradeOnAppend(
```

Comment:
could be renamed to getExtraVersionsForAppendSegments

Reply:
"extra" versions is probably still confusing. How about get `createUpgradedVersionsOfAppendSegments`?

Reply:
or newVersionsOfAppendSegments

Reply:
Yeah, that works too. But I just want to avoid confusion between words like "new", "upgrade" or "extra" when all of them mean the same thing in our context.